cmd/fs-v1-metadata.go

```diff
@@ -315,37 +316,55 @@ func createFormatFS(fsPath string) error {
 }
 
 func initFormatFS(fsPath string) (rlk *lock.RLockedFile, err error) {
+	// Done channel is used to close any lingering retry routine, as soon
+	// as this function returns.
+	doneCh := make(chan struct{})
+
+	// Indicate to our retry routine to exit cleanly, upon this function return.
+	defer close(doneCh)
+
 	// This loop validates format.json by holding a read lock and
 	// proceeds if disk unformatted to hold non-blocking WriteLock
 	// If for some reason non-blocking WriteLock fails and the error
 	// is lock.ErrAlreadyLocked i.e some other process is holding a
 	// lock we retry in the loop again.
+	console.Printf("Initializing FS backend format.. ")
+
+	// Wait on the jitter retry loop.
+	retryTimerCh := newRetryTimerSimple(doneCh)
 	for {
-		// Validate the `format.json` for expected values.
-		rlk, err = checkLockedValidFormatFS(fsPath)
-		switch {
-		case err == nil:
-			// Holding a read lock ensures that any write lock operation
-			// is blocked if attempted in-turn avoiding corruption on
-			// the backend disk.
-			return rlk, nil
-		case errorCause(err) == errUnformattedDisk:
-			if err = createFormatFS(fsPath); err != nil {
-				// Existing write locks detected.
-				if errorCause(err) == lock.ErrAlreadyLocked {
-					// Lock already present, sleep and attempt again.
-					time.Sleep(100 * time.Millisecond)
-					continue
+		select {
+		case _ = <-retryTimerCh:
```

Comment:
If we receive more than once on this channel then we will end up calling checkLockedValidFormatFS() more than once.